R/open.R

```diff
@@ -1,40 +1,57 @@
 #' Open an RStudio Project from a repository
-#' 
+#'
 #' @param repo A character vector of length one with a repository name. Fixed
-#'   regex (i.e. substring matches) are allowed.
-#' @param n The number upper bound of repos to open. If `NULL`, all will be 
-#' opened.
-#' @param project A character vector of length one with a project name. 
-#'   Can be left `NULL` if the repo name is unique accros all projects.
+#' regex (i.e. substring matches) are allowed.
+#' @param n For partial matches, the ID of repos to open in parallel. Can be a
+#'   single integer, a sequence (`3:7`) or a vector (`c(2, 5)`). For `n = 1`,
+#'   the exact match is opened (if there is one) otherwise the best regex fit is
+#'   opened.
+#' @param project A character vector of length one with a project name.
+#'   Can be left `NULL` if the repo name is unique across all projects.
 #' @param dir Any directory under the team root.
-#' @param newSession whether or not to open the RStudio Project in a new 
+#' @param newSession whether or not to open the RStudio Project in a new
 #'   session.
-#' @importFrom rlang abort 
+#' @importFrom rlang abort
 #' @importFrom purrr walk
 #' @export
-open_repo <- function(repo = "meta", 
+open_repo <- function(repo = "meta",
                       n = 1,
-                      project = NULL, 
-                      dir = ".", 
+                      project = NULL,
+                      dir = ".",
                       newSession = TRUE) {
+
+
   repos <- git_dirs(dir)
   is_repo_match <- grepl(repo, basename(repos), fixed = TRUE)
   matching_repo <- repos[is_repo_match]
-  
+
   if (all(!is_repo_match)) stop("Make sure the repo exists.", call. = FALSE)
   if (!is.null(project)) {
     is_project_match <- basename(dirname(repos)) == project
     matching_repo <- repos[is_repo_match & is_project_match]
   }
-  
+
   if (!is.null(n) && sum(is_repo_match) > n) {
```

Comment:
if `n` is a vector, what happens here?

Reply:
Also, the idea of `n` was to have an upper bound to not accidentally open 100 repos, which is why the code previously returned an error when the condition was met.

Reply:
Feel free to polish, for me it works now as wished. :)
I can open a direct match and multiple by supplying a vector.

Reply:
Ok, I might twist it a bit later, hopefully today. Thanks for you work on extending it.